tests/jballerina-unit-test/src/test/resources/test-src/statements/matchstmt/mapping-match-pattern.bal

```diff
@@ -598,6 +598,28 @@ function testMappingMatchPattern26() {
     assertEquals(1, mappingMatchPattern27({y: "hello world", x: 1}));
 }
 
+
+type T (map<string> & readonly) | map<int>;
+
+function mappingMatchPattern28(T v) returns string {
```

Comment:
Also works?

It is OK to have these tests also, but we need to make sure we have tests covering the previously uncovered path, and that those test would fail without this fix.